--
-- PostgreSQL database dump
--

-- Dumped from database version 15.10 (Debian 15.10-1.pgdg120+1)
-- Dumped by pg_dump version 15.10 (Debian 15.10-1.pgdg120+1)

SET statement_timeout = 0;
SET lock_timeout = 0;
SET idle_in_transaction_session_timeout = 0;
SET client_encoding = 'UTF8';
SET standard_conforming_strings = on;
SELECT pg_catalog.set_config('search_path', '', false);
SET check_function_bodies = false;
SET xmloption = content;
SET client_min_messages = warning;
SET row_security = off;

SET default_tablespace = '';

SET default_table_access_method = heap;

--
-- Name: plugins; Type: TABLE; Schema: public; Owner: testdb
--

CREATE TABLE public.plugins (
    created_at timestamp without time zone,
    updated_at timestamp without time zone,
    id integer NOT NULL,
    name character varying NOT NULL,
    description character varying NOT NULL,
    author character varying NOT NULL,
    plugin_path character varying NOT NULL,
    dependencies jsonb,
    drawflow jsonb NOT NULL,
    rules jsonb NOT NULL
);


ALTER TABLE public.plugins OWNER TO testdb;

--
-- Name: plugins_id_seq; Type: SEQUENCE; Schema: public; Owner: testdb
--

CREATE SEQUENCE public.plugins_id_seq
    AS integer
    START WITH 1
    INCREMENT BY 1
    NO MINVALUE
    NO MAXVALUE
    CACHE 1;


ALTER TABLE public.plugins_id_seq OWNER TO testdb;

--
-- Name: plugins_id_seq; Type: SEQUENCE OWNED BY; Schema: public; Owner: testdb
--

ALTER SEQUENCE public.plugins_id_seq OWNED BY public.plugins.id;


--
-- Name: plugins id; Type: DEFAULT; Schema: public; Owner: testdb
--

ALTER TABLE ONLY public.plugins ALTER COLUMN id SET DEFAULT nextval('public.plugins_id_seq'::regclass);


--
-- Data for Name: plugins; Type: TABLE DATA; Schema: public; Owner: testdb
--

INSERT INTO public.plugins (created_at, updated_at, id, name, description, author, plugin_path, dependencies, drawflow, rules) VALUES ('2024-10-07 18:35:39.920611', '2024-11-23 14:53:30.033342', 1, 'FastTENET', 'FastTENET is an accelerated TENET algorithm based on manycore computing.', 'test', './plugin/FastTENET/', '{"renv.lock": "{\n  \"R\": {\n    \"Version\": \"4.4.1\",\n    \"Repositories\": [\n      {\n        \"Name\": \"CRAN\",\n        \"URL\": \"https://cran.yu.ac.kr\"\n      }\n    ]\n  },\n  \"Packages\": {\n    \"ggplot2\": {\n      \"Package\": \"ggplot2\",\n      \"Version\": \"3.5.1\",\n      \"Source\": \"Repository\",\n      \"Repository\": \"CRAN\",\n      \"Requirements\": [\n        \"MASS\",\n        \"R\",\n        \"cli\",\n        \"glue\",\n        \"grDevices\",\n        \"grid\",\n        \"gtable\",\n        \"isoband\",\n        \"lifecycle\",\n        \"mgcv\",\n        \"rlang\",\n        \"scales\",\n        \"stats\",\n        \"tibble\",\n        \"vctrs\",\n        \"withr\"\n      ],\n      \"Hash\": \"44c6a2f8202d5b7e878ea274b1092426\"\n    },\n    \"pheatmap\": {\n      \"Package\": \"pheatmap\",\n      \"Version\": \"1.0.12\",\n      \"Source\": \"Repository\",\n      \"Repository\": \"CRAN\",\n      \"Requirements\": [\n        \"R\",\n        \"RColorBrewer\",\n        \"grDevices\",\n        \"graphics\",\n        \"grid\",\n        \"gtable\",\n        \"scales\",\n        \"stats\"\n      ],\n      \"Hash\": \"db1fb0021811b6693741325bbe916e58\"\n    },\n    \"plotly\": {\n      \"Package\": \"plotly\",\n      \"Version\": \"4.10.4\",\n      \"Source\": \"Repository\",\n      \"Repository\": \"CRAN\",\n      \"Requirements\": [\n        \"R\",\n        \"RColorBrewer\",\n        \"base64enc\",\n        \"crosstalk\",\n        \"data.table\",\n        \"digest\",\n        \"dplyr\",\n        \"ggplot2\",\n        \"htmltools\",\n        \"htmlwidgets\",\n        \"httr\",\n        \"jsonlite\",\n        \"lazyeval\",\n        \"magrittr\",\n        \"promises\",\n        \"purrr\",\n        \"rlang\",\n        \"scales\",\n        \"tibble\",\n        \"tidyr\",\n        \"tools\",\n        \"vctrs\",\n        \"viridisLite\"\n      ],\n      \"Hash\": \"a1ac5c03ad5ad12b9d1597e00e23c3dd\"\n    },\n    \"jsonlite\": {\n      \"Package\": \"jsonlite\",\n      \"Version\": \"1.8.8\",\n      \"Source\": \"Repository\",\n      \"Repository\": \"CRAN\",\n      \"Requirements\": [\n        \"methods\"\n      ],\n      \"Hash\": \"e1b9c55281c5adc4dd113652d9e26768\"\n    }\n  }\n}\n", "requirements.txt": "fasttenet==0.1.9\nnetworkx>=3.0\nmatplotlib>=3.5.2\nplotly>=5.22.0"}', '{"drawflow": {"Home": {"data": {"19": {"id": 19, "data": {"title": "Heatmap", "inputs": ["expression.csv", "trajectory.txt", "target.sif"], "script": {}, "outputs": ["Heatmap.json"], "parameters": [{"max": null, "min": null, "name": "expression", "type": "inputFile", "defaultValue": "expression.csv", "fileExtension": ".csv"}, {"max": null, "min": null, "name": "trajectory", "type": "inputFile", "defaultValue": "trajectory.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "target", "type": "inputFile", "defaultValue": "target.sif", "fileExtension": ".sif"}, {"name": "output", "type": "outputFile", "defaultValue": "Heatmap.json", "fileExtension": ".json"}, {"max": null, "min": 5, "name": "Top Genes", "type": "int", "defaultValue": "10", "fileExtension": ""}, {"max": null, "min": 50, "name": "Sample Size", "type": "int", "defaultValue": "100", "fileExtension": ""}], "isVisualization": true}, "html": "ruleNode", "name": "ruleNode", "class": "visualizationNode", "pos_x": 1056, "pos_y": 190, "inputs": {"input_1": {"connections": []}, "input_2": {"connections": []}, "input_3": {"connections": []}}, "outputs": {"output_1": {"connections": []}}, "typenode": "vue"}, "20": {"id": 20, "data": {"title": "Barplot", "inputs": ["target.sif"], "script": {}, "outputs": ["Barplot.json"], "parameters": [{"max": null, "min": null, "name": "target", "type": "inputFile", "defaultValue": "target.sif", "fileExtension": ".sif"}, {"name": "output", "type": "outputFile", "defaultValue": "Barplot.json", "fileExtension": ".json"}, {"max": null, "min": 5, "name": "Top Genes", "type": "int", "defaultValue": "10", "fileExtension": ""}], "isVisualization": true}, "html": "ruleNode", "name": "ruleNode", "class": "visualizationNode", "pos_x": 847, "pos_y": -7, "inputs": {"input_1": {"connections": []}}, "outputs": {"output_1": {"connections": []}}, "typenode": "vue"}, "21": {"id": 21, "data": {"title": "Network", "inputs": ["target.sif"], "script": {}, "outputs": ["Network.json"], "parameters": [{"max": null, "min": null, "name": "target", "type": "inputFile", "defaultValue": "target.sif", "fileExtension": ".sif"}, {"name": "output", "type": "outputFile", "defaultValue": "Network.json", "fileExtension": ".json"}, {"max": null, "min": 10, "name": "Top Genes", "type": "int", "defaultValue": "50", "fileExtension": ""}], "isVisualization": true}, "html": "ruleNode", "name": "ruleNode", "class": "visualizationNode", "pos_x": 904, "pos_y": 78, "inputs": {"input_1": {"connections": []}}, "outputs": {"output_1": {"connections": []}}, "typenode": "vue"}, "22": {"id": 22, "data": {"title": "Run", "inputs": ["expression.csv", "trajectory.txt", "cellSelect.txt", "TF.txt"], "script": {}, "outputs": ["resultMatrix.txt"], "parameters": [{"max": null, "min": null, "name": "expression", "type": "inputFile", "defaultValue": "expression.csv", "fileExtension": ".csv"}, {"max": null, "min": null, "name": "trajectory", "type": "inputFile", "defaultValue": "trajectory.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "cellSelect", "type": "inputFile", "defaultValue": "cellSelect.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "TF", "type": "optionalInputFile", "defaultValue": "TF.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "resultMatrix", "type": "outputFile", "defaultValue": "resultMatrix.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "backend", "type": "string", "defaultValue": "gpu", "fileExtension": ""}, {"max": 16, "min": 1, "name": "number of devices", "type": "int", "defaultValue": "4", "fileExtension": ""}, {"max": null, "min": 2, "name": "batch size", "type": "int", "defaultValue": "32768", "fileExtension": ""}], "isVisualization": false}, "html": "ruleNode", "name": "ruleNode", "class": "ruleNode", "pos_x": 534, "pos_y": 45, "inputs": {"input_1": {"connections": []}, "input_2": {"connections": []}, "input_3": {"connections": []}, "input_4": {"connections": []}}, "outputs": {"output_1": {"connections": [{"node": "23", "output": "input_1"}]}}, "typenode": "vue"}, "23": {"id": 23, "data": {"title": "Make GRN", "inputs": ["resultMatrix.txt", "expression.csv", "TF.txt"], "script": {}, "outputs": ["resultMatrixFDR.sif", "resultMatrixFdrOutdegree.txt", "resultMatrixFdrTrim.sif", "resultMatrixFdrTrimOutdegree.txt"], "parameters": [{"max": null, "min": null, "name": "resultMatrix", "type": "inputFile", "defaultValue": "resultMatrix.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "expression", "type": "inputFile", "defaultValue": "expression.csv", "fileExtension": ".csv"}, {"max": null, "min": null, "name": "TF", "type": "optionalInputFile", "defaultValue": "TF.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "resultMatrixFDR", "type": "outputFile", "defaultValue": "resultMatrixFDR.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "resultMatrixFdrOutdegree", "type": "outputFile", "defaultValue": "resultMatrixFdrOutdegree.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "resultMatrixFdrTrim", "type": "outputFile", "defaultValue": "resultMatrixFdrTrim.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "resultMatrixFdrTrimOutdegree", "type": "outputFile", "defaultValue": "resultMatrixFdrTrimOutdegree.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "FDR", "type": "float", "defaultValue": "0.01", "fileExtension": ""}, {"max": null, "min": 0, "name": "number of outdegrees", "type": "int", "defaultValue": "0", "fileExtension": ""}, {"max": null, "min": null, "name": "trimming threshold", "type": "float", "defaultValue": "0", "fileExtension": ""}], "isVisualization": false}, "html": "ruleNode", "name": "ruleNode", "class": "ruleNode", "pos_x": 799, "pos_y": 165, "inputs": {"input_1": {"connections": [{"node": "22", "input": "output_1"}]}, "input_2": {"connections": []}, "input_3": {"connections": []}}, "outputs": {"output_1": {"connections": []}, "output_2": {"connections": []}, "output_3": {"connections": []}, "output_4": {"connections": []}}, "typenode": "vue"}}}}}', '{"0": {"name": "Run", "input": ["expression.csv", "trajectory.txt", "cellSelect.txt", "TF.txt"], "nodeId": 22, "output": ["resultMatrix.txt"], "script": "run_optional_tf.py", "parameters": [{"max": null, "min": null, "name": "expression", "type": "inputFile", "defaultValue": "expression.csv", "fileExtension": ".csv"}, {"max": null, "min": null, "name": "trajectory", "type": "inputFile", "defaultValue": "trajectory.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "cellSelect", "type": "inputFile", "defaultValue": "cellSelect.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "TF", "type": "optionalInputFile", "defaultValue": "TF.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "resultMatrix", "type": "outputFile", "defaultValue": "resultMatrix.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "backend", "type": "string", "defaultValue": "gpu", "fileExtension": ""}, {"max": 16, "min": 1, "name": "number of devices", "type": "int", "defaultValue": "4", "fileExtension": ""}, {"max": null, "min": 2, "name": "batch size", "type": "int", "defaultValue": "32768", "fileExtension": ""}], "isVisualization": false}, "1": {"name": "Heatmap", "input": ["expression.csv", "trajectory.txt", "target.sif"], "nodeId": 19, "output": ["Heatmap.json"], "script": "Pseudotime_heatmap.R", "parameters": [{"max": null, "min": null, "name": "expression", "type": "inputFile", "defaultValue": "expression.csv", "fileExtension": ".csv"}, {"max": null, "min": null, "name": "trajectory", "type": "inputFile", "defaultValue": "trajectory.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "target", "type": "inputFile", "defaultValue": "target.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "output", "type": "outputFile", "defaultValue": "Heatmap.json", "fileExtension": ".json"}, {"max": null, "min": 5, "name": "Top Genes", "type": "int", "defaultValue": "10", "fileExtension": ""}, {"max": null, "min": 50, "name": "Sample Size", "type": "int", "defaultValue": "100", "fileExtension": ""}], "isVisualization": true}, "2": {"name": "Barplot", "input": ["target.sif"], "nodeId": 20, "output": ["Barplot.json"], "script": "Barplot.py", "parameters": [{"max": null, "min": null, "name": "target", "type": "inputFile", "defaultValue": "target.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "output", "type": "outputFile", "defaultValue": "Barplot.json", "fileExtension": ".json"}, {"max": null, "min": 5, "name": "Top Genes", "type": "int", "defaultValue": "10", "fileExtension": ""}], "isVisualization": true}, "3": {"name": "Network", "input": ["target.sif"], "nodeId": 21, "output": ["Network.json"], "script": "NetworkX.py", "parameters": [{"max": null, "min": null, "name": "target", "type": "inputFile", "defaultValue": "target.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "output", "type": "outputFile", "defaultValue": "Network.json", "fileExtension": ".json"}, {"max": null, "min": 10, "name": "Top Genes", "type": "int", "defaultValue": "50", "fileExtension": ""}], "isVisualization": true}, "4": {"name": "Make GRN", "input": ["resultMatrix.txt", "expression.csv", "TF.txt"], "nodeId": 23, "output": ["resultMatrixFDR.sif", "resultMatrixFdrOutdegree.txt", "resultMatrixFdrTrim.sif", "resultMatrixFdrTrimOutdegree.txt"], "script": "make_grn.py", "parameters": [{"max": null, "min": null, "name": "resultMatrix", "type": "inputFile", "defaultValue": "resultMatrix.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "expression", "type": "inputFile", "defaultValue": "expression.csv", "fileExtension": ".csv"}, {"max": null, "min": null, "name": "TF", "type": "optionalInputFile", "defaultValue": "TF.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "resultMatrixFDR", "type": "outputFile", "defaultValue": "resultMatrixFDR.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "resultMatrixFdrOutdegree", "type": "outputFile", "defaultValue": "resultMatrixFdrOutdegree.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "resultMatrixFdrTrim", "type": "outputFile", "defaultValue": "resultMatrixFdrTrim.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "resultMatrixFdrTrimOutdegree", "type": "outputFile", "defaultValue": "resultMatrixFdrTrimOutdegree.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "FDR", "type": "float", "defaultValue": "0.01", "fileExtension": ""}, {"max": null, "min": 0, "name": "number of outdegrees", "type": "int", "defaultValue": "0", "fileExtension": ""}, {"max": null, "min": null, "name": "trimming threshold", "type": "float", "defaultValue": "0", "fileExtension": ""}], "isVisualization": false}}');
INSERT INTO public.plugins (created_at, updated_at, id, name, description, author, plugin_path, dependencies, drawflow, rules) VALUES ('2024-10-07 19:00:00.355031', '2024-11-23 15:57:57.455602', 2, 'TENET', 'A tool for reconstructing Transfer Entropy-based causal gene NETwork from pseudo-time ordered single cell transcriptomic data', 'test', './plugin/TENET/', '{"renv.lock": "{\n  \"R\": {\n    \"Version\": \"4.4.1\",\n    \"Repositories\": [\n      {\n        \"Name\": \"CRAN\",\n        \"URL\": \"https://cran.yu.ac.kr\"\n      }\n    ]\n  },\n  \"Packages\": {\n    \"ggplot2\": {\n      \"Package\": \"ggplot2\",\n      \"Version\": \"3.5.1\",\n      \"Source\": \"Repository\",\n      \"Repository\": \"CRAN\",\n      \"Requirements\": [\n        \"MASS\",\n        \"R\",\n        \"cli\",\n        \"glue\",\n        \"grDevices\",\n        \"grid\",\n        \"gtable\",\n        \"isoband\",\n        \"lifecycle\",\n        \"mgcv\",\n        \"rlang\",\n        \"scales\",\n        \"stats\",\n        \"tibble\",\n        \"vctrs\",\n        \"withr\"\n      ],\n      \"Hash\": \"44c6a2f8202d5b7e878ea274b1092426\"\n    },\n    \"pheatmap\": {\n      \"Package\": \"pheatmap\",\n      \"Version\": \"1.0.12\",\n      \"Source\": \"Repository\",\n      \"Repository\": \"CRAN\",\n      \"Requirements\": [\n        \"R\",\n        \"RColorBrewer\",\n        \"grDevices\",\n        \"graphics\",\n        \"grid\",\n        \"gtable\",\n        \"scales\",\n        \"stats\"\n      ],\n      \"Hash\": \"db1fb0021811b6693741325bbe916e58\"\n    },\n    \"plotly\": {\n      \"Package\": \"plotly\",\n      \"Version\": \"4.10.4\",\n      \"Source\": \"Repository\",\n      \"Repository\": \"CRAN\",\n      \"Requirements\": [\n        \"R\",\n        \"RColorBrewer\",\n        \"base64enc\",\n        \"crosstalk\",\n        \"data.table\",\n        \"digest\",\n        \"dplyr\",\n        \"ggplot2\",\n        \"htmltools\",\n        \"htmlwidgets\",\n        \"httr\",\n        \"jsonlite\",\n        \"lazyeval\",\n        \"magrittr\",\n        \"promises\",\n        \"purrr\",\n        \"rlang\",\n        \"scales\",\n        \"tibble\",\n        \"tidyr\",\n        \"tools\",\n        \"vctrs\",\n        \"viridisLite\"\n      ],\n      \"Hash\": \"a1ac5c03ad5ad12b9d1597e00e23c3dd\"\n    },\n    \"jsonlite\": {\n      \"Package\": \"jsonlite\",\n      \"Version\": \"1.8.8\",\n      \"Source\": \"Repository\",\n      \"Repository\": \"CRAN\",\n      \"Requirements\": [\n        \"methods\"\n      ],\n      \"Hash\": \"e1b9c55281c5adc4dd113652d9e26768\"\n    }\n  }\n}\n", "requirements.txt": "fasttenet==0.1.7\nnetworkx>=3.0\nmatplotlib>=3.5.2\nplotly>=5.22.0"}', '{"drawflow": {"Home": {"data": {"16": {"id": 16, "data": {"title": "GRN Reconstruction : FDR", "inputs": ["resultMatrix.txt", "TF.txt"], "script": {}, "outputs": ["resultMatrixFdr.sif"], "parameters": [{"max": null, "min": null, "name": "resultMatrix", "type": "inputFile", "defaultValue": "resultMatrix.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "TF", "type": "optionalInputFile", "defaultValue": "TF.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "resultMatrixFdr", "type": "outputFile", "defaultValue": "resultMatrixFdr.sif", "fileExtension": ".sif"}, {"max": 1, "min": null, "name": "FDR", "type": "float", "defaultValue": "0.01", "fileExtension": ""}], "isVisualization": false}, "html": "ruleNode", "name": "ruleNode", "class": "ruleNode", "pos_x": 737, "pos_y": 70, "inputs": {"input_1": {"connections": [{"node": "33", "input": "output_1"}]}, "input_2": {"connections": []}}, "outputs": {"output_1": {"connections": [{"node": "29", "output": "input_1"}, {"node": "30", "output": "input_1"}]}}, "typenode": "vue"}, "19": {"id": 19, "data": {"title": "NeworkX", "inputs": ["target.sif"], "script": {}, "outputs": ["NeworkX.json"], "parameters": [{"max": null, "min": null, "name": "target", "type": "inputFile", "defaultValue": "target.sif", "fileExtension": ".sif"}, {"name": "output", "type": "outputFile", "defaultValue": "NeworkX.json", "fileExtension": ".json"}, {"max": 200, "min": 10, "name": "# Top Genes", "type": "int", "defaultValue": "50", "fileExtension": ""}], "isVisualization": true}, "html": "ruleNode", "name": "ruleNode", "class": "visualizationNode", "pos_x": 1224, "pos_y": 111, "inputs": {"input_1": {"connections": []}}, "outputs": {"output_1": {"connections": []}}, "typenode": "vue"}, "20": {"id": 20, "data": {"title": "Barplot", "inputs": ["target.sif"], "script": {}, "outputs": ["Barplot.json"], "parameters": [{"max": null, "min": null, "name": "target", "type": "inputFile", "defaultValue": "target.sif", "fileExtension": ".sif"}, {"name": "output", "type": "outputFile", "defaultValue": "Barplot.json", "fileExtension": ".json"}, {"max": 100, "min": 5, "name": "# Top Genes", "type": "int", "defaultValue": "10", "fileExtension": ""}], "isVisualization": true}, "html": "ruleNode", "name": "ruleNode", "class": "visualizationNode", "pos_x": 859, "pos_y": 208, "inputs": {"input_1": {"connections": []}}, "outputs": {"output_1": {"connections": []}}, "typenode": "vue"}, "21": {"id": 21, "data": {"title": "Heatmap", "inputs": ["expMatrix.csv", "pseudotime.csv", "target.sif"], "script": {}, "outputs": ["Heatmap.json"], "parameters": [{"max": null, "min": null, "name": "expMatrix", "type": "inputFile", "defaultValue": "expMatrix.csv", "fileExtension": ".csv"}, {"max": null, "min": null, "name": "pseudotime", "type": "inputFile", "defaultValue": "pseudotime.csv", "fileExtension": ".csv"}, {"max": null, "min": null, "name": "target", "type": "inputFile", "defaultValue": "target.sif", "fileExtension": ".sif"}, {"name": "output", "type": "outputFile", "defaultValue": "Heatmap.json", "fileExtension": ".json"}, {"max": null, "min": 5, "name": "# Top Genes", "type": "int", "defaultValue": "10", "fileExtension": ""}, {"max": 1000, "min": 50, "name": "Sample Size", "type": "int", "defaultValue": "100", "fileExtension": ""}], "isVisualization": true}, "html": "ruleNode", "name": "ruleNode", "class": "visualizationNode", "pos_x": 1139, "pos_y": 237, "inputs": {"input_1": {"connections": [{"node": "32", "input": "output_1"}]}, "input_2": {"connections": [{"node": "32", "input": "output_2"}]}, "input_3": {"connections": []}}, "outputs": {"output_1": {"connections": []}}, "typenode": "vue"}, "28": {"id": 28, "data": {"title": "GRN Reconstruction : NumLinks", "inputs": ["resultMatrix.txt", "TF.txt"], "script": {}, "outputs": ["resultMatrixNumlinks.sif"], "parameters": [{"max": null, "min": null, "name": "resultMatrix", "type": "inputFile", "defaultValue": "resultMatrix.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "TF", "type": "optionalInputFile", "defaultValue": "TF.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "resultMatrixNumlinks", "type": "outputFile", "defaultValue": "resultMatrixNumlinks.sif", "fileExtension": ".sif"}, {"max": null, "min": 100, "name": "number of links", "type": "int", "defaultValue": "1000", "fileExtension": ""}], "isVisualization": false}, "html": "ruleNode", "name": "ruleNode", "class": "ruleNode", "pos_x": 742, "pos_y": -25, "inputs": {"input_1": {"connections": [{"node": "33", "input": "output_1"}]}, "input_2": {"connections": []}}, "outputs": {"output_1": {"connections": [{"node": "29", "output": "input_2"}, {"node": "30", "output": "input_2"}]}}, "typenode": "vue"}, "29": {"id": 29, "data": {"title": "Indirect Edge Trimming", "inputs": ["resultMatrixFdr.sif", "resultMatrixNumlinks.sif"], "script": {}, "outputs": ["fdrTrim.sif", "numlinksTrim.sif"], "parameters": [{"max": null, "min": null, "name": "resultMatrixFdr", "type": "inputFile", "defaultValue": "resultMatrixFdr.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "resultMatrixNumlinks", "type": "inputFile", "defaultValue": "resultMatrixNumlinks.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "fdrTrim", "type": "outputFile", "defaultValue": "fdrTrim.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "numlinksTrim", "type": "outputFile", "defaultValue": "numlinksTrim.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "Cutoff", "type": "float", "defaultValue": "0", "fileExtension": ""}], "isVisualization": false}, "html": "ruleNode", "name": "ruleNode", "class": "ruleNode", "pos_x": 981, "pos_y": 14, "inputs": {"input_1": {"connections": [{"node": "16", "input": "output_1"}]}, "input_2": {"connections": [{"node": "28", "input": "output_1"}]}}, "outputs": {"output_1": {"connections": [{"node": "30", "output": "input_3"}]}, "output_2": {"connections": [{"node": "30", "output": "input_4"}]}}, "typenode": "vue"}, "30": {"id": 30, "data": {"title": "Counting Outdegree", "inputs": ["resultMatrixFdr.sif", "resultMatrixNumlinks.sif", "fdrTrim.sif", "numlinksTrim.sif"], "script": {}, "outputs": ["fdrOutdegree.txt", "numlinksOutdegree.txt", "fdrTrimOutdegree.txt", "numlinksTrimOutdegree.txt"], "parameters": [{"max": null, "min": null, "name": "resultMatrixFdr", "type": "inputFile", "defaultValue": "resultMatrixFdr.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "resultMatrixNumlinks", "type": "inputFile", "defaultValue": "resultMatrixNumlinks.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "fdrTrim", "type": "inputFile", "defaultValue": "fdrTrim.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "numlinksTrim", "type": "inputFile", "defaultValue": "numlinksTrim.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "fdrOutdegree", "type": "outputFile", "defaultValue": "fdrOutdegree.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "numlinksOutdegree", "type": "outputFile", "defaultValue": "numlinksOutdegree.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "fdrTrimOutdegree", "type": "outputFile", "defaultValue": "fdrTrimOutdegree.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "numlinksTrimOutdegree", "type": "outputFile", "defaultValue": "numlinksTrimOutdegree.txt", "fileExtension": ".txt"}], "isVisualization": false}, "html": "ruleNode", "name": "ruleNode", "class": "ruleNode", "pos_x": 1231, "pos_y": -83, "inputs": {"input_1": {"connections": [{"node": "16", "input": "output_1"}]}, "input_2": {"connections": [{"node": "28", "input": "output_1"}]}, "input_3": {"connections": [{"node": "29", "input": "output_1"}]}, "input_4": {"connections": [{"node": "29", "input": "output_2"}]}}, "outputs": {"output_1": {"connections": []}, "output_2": {"connections": []}, "output_3": {"connections": []}, "output_4": {"connections": []}}, "typenode": "vue"}, "32": {"id": 32, "data": {"title": "TENET Input", "inputs": ["input.h5ad", "geneList.txt"], "script": {}, "outputs": ["expMatrix.csv", "pseudotime.csv", "cellSelect.csv"], "parameters": [{"max": null, "min": null, "name": "input", "type": "inputFile", "defaultValue": "input.h5ad", "fileExtension": ".h5ad"}, {"max": null, "min": null, "name": "geneList", "type": "optionalInputFile", "defaultValue": "geneList.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "ScatterPlot", "type": "string", "defaultValue": "", "fileExtension": ""}, {"max": null, "min": null, "name": "expMatrix", "type": "outputFile", "defaultValue": "expMatrix.csv", "fileExtension": ".csv"}, {"max": null, "min": null, "name": "pseudotime", "type": "outputFile", "defaultValue": "pseudotime.csv", "fileExtension": ".csv"}, {"max": null, "min": null, "name": "cellSelect", "type": "outputFile", "defaultValue": "cellSelect.csv", "fileExtension": ".csv"}, {"max": null, "min": null, "name": "cell group", "type": "h5adParameter", "defaultValue": "", "fileExtension": ""}, {"max": null, "min": null, "name": "pseudotime", "type": "h5adParameter", "defaultValue": "", "fileExtension": ""}, {"max": null, "min": null, "name": "clusters", "type": "h5adParameter", "defaultValue": "", "fileExtension": ""}], "isVisualization": false}, "html": "ruleNode", "name": "ruleNode", "class": "ruleNode", "pos_x": 133, "pos_y": 69, "inputs": {"input_1": {"connections": []}, "input_2": {"connections": []}}, "outputs": {"output_1": {"connections": [{"node": "21", "output": "input_1"}, {"node": "33", "output": "input_1"}]}, "output_2": {"connections": [{"node": "21", "output": "input_2"}, {"node": "33", "output": "input_2"}]}, "output_3": {"connections": [{"node": "33", "output": "input_3"}]}}, "typenode": "vue"}, "33": {"id": 33, "data": {"title": "Run", "inputs": ["expMatrix.csv", "pseudotime.csv", "cellSelect.csv", "TF.txt"], "script": {}, "outputs": ["resultMatrix.txt"], "parameters": [{"max": null, "min": null, "name": "expMatrix", "type": "inputFile", "defaultValue": "expMatrix.csv", "fileExtension": ".csv"}, {"max": null, "min": null, "name": "pseudotime", "type": "inputFile", "defaultValue": "pseudotime.csv", "fileExtension": ".csv"}, {"max": null, "min": null, "name": "cellSelect", "type": "inputFile", "defaultValue": "cellSelect.csv", "fileExtension": ".csv"}, {"max": null, "min": null, "name": "TF", "type": "optionalInputFile", "defaultValue": "TF.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "resultMatrix", "type": "outputFile", "defaultValue": "resultMatrix.txt", "fileExtension": ".txt"}, {"max": 32, "min": 1, "name": "number of devices", "type": "int", "defaultValue": "4", "fileExtension": ""}], "isVisualization": false}, "html": "ruleNode", "name": "ruleNode", "class": "ruleNode", "pos_x": 470, "pos_y": 80, "inputs": {"input_1": {"connections": [{"node": "32", "input": "output_1"}]}, "input_2": {"connections": [{"node": "32", "input": "output_2"}]}, "input_3": {"connections": [{"node": "32", "input": "output_3"}]}, "input_4": {"connections": []}}, "outputs": {"output_1": {"connections": [{"node": "16", "output": "input_1"}, {"node": "28", "output": "input_1"}]}}, "typenode": "vue"}}}}}', '{"0": {"name": "TENET Input", "input": ["input.h5ad", "geneList.txt"], "nodeId": 32, "output": ["expMatrix.csv", "pseudotime.csv", "cellSelect.csv"], "script": "tenet_input.py", "parameters": [{"max": null, "min": null, "name": "input", "type": "inputFile", "defaultValue": "input.h5ad", "fileExtension": ".h5ad"}, {"max": null, "min": null, "name": "geneList", "type": "optionalInputFile", "defaultValue": "geneList.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "ScatterPlot", "type": "string", "defaultValue": "", "fileExtension": ""}, {"max": null, "min": null, "name": "expMatrix", "type": "outputFile", "defaultValue": "expMatrix.csv", "fileExtension": ".csv"}, {"max": null, "min": null, "name": "pseudotime", "type": "outputFile", "defaultValue": "pseudotime.csv", "fileExtension": ".csv"}, {"max": null, "min": null, "name": "cellSelect", "type": "outputFile", "defaultValue": "cellSelect.csv", "fileExtension": ".csv"}, {"max": null, "min": null, "name": "cell group", "type": "h5adParameter", "defaultValue": "", "fileExtension": ""}, {"max": null, "min": null, "name": "pseudotime", "type": "h5adParameter", "defaultValue": "", "fileExtension": ""}, {"max": null, "min": null, "name": "clusters", "type": "h5adParameter", "defaultValue": "", "fileExtension": ""}], "isVisualization": false}, "1": {"name": "GRN Reconstruction : FDR", "input": ["resultMatrix.txt", "TF.txt"], "nodeId": 16, "output": ["resultMatrixFdr.sif"], "script": "makeGRNbyTF.py", "parameters": [{"max": null, "min": null, "name": "resultMatrix", "type": "inputFile", "defaultValue": "resultMatrix.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "TF", "type": "optionalInputFile", "defaultValue": "TF.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "resultMatrixFdr", "type": "outputFile", "defaultValue": "resultMatrixFdr.sif", "fileExtension": ".sif"}, {"max": 1, "min": null, "name": "FDR", "type": "float", "defaultValue": "0.01", "fileExtension": ""}], "isVisualization": false}, "2": {"name": "GRN Reconstruction : NumLinks", "input": ["resultMatrix.txt", "TF.txt"], "nodeId": 28, "output": ["resultMatrixNumlinks.sif"], "script": "makeGRNbyTFsameNumberOfLinks.py", "parameters": [{"max": null, "min": null, "name": "resultMatrix", "type": "inputFile", "defaultValue": "resultMatrix.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "TF", "type": "optionalInputFile", "defaultValue": "TF.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "resultMatrixNumlinks", "type": "outputFile", "defaultValue": "resultMatrixNumlinks.sif", "fileExtension": ".sif"}, {"max": null, "min": 100, "name": "number of links", "type": "int", "defaultValue": "1000", "fileExtension": ""}], "isVisualization": false}, "3": {"name": "Indirect Edge Trimming", "input": ["resultMatrixFdr.sif", "resultMatrixNumlinks.sif"], "nodeId": 29, "output": ["fdrTrim.sif", "numlinksTrim.sif"], "script": "trim_indirect.py", "parameters": [{"max": null, "min": null, "name": "resultMatrixFdr", "type": "inputFile", "defaultValue": "resultMatrixFdr.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "resultMatrixNumlinks", "type": "inputFile", "defaultValue": "resultMatrixNumlinks.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "fdrTrim", "type": "outputFile", "defaultValue": "fdrTrim.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "numlinksTrim", "type": "outputFile", "defaultValue": "numlinksTrim.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "Cutoff", "type": "float", "defaultValue": "0", "fileExtension": ""}], "isVisualization": false}, "4": {"name": "Counting Outdegree", "input": ["resultMatrixFdr.sif", "resultMatrixNumlinks.sif", "fdrTrim.sif", "numlinksTrim.sif"], "nodeId": 30, "output": ["fdrOutdegree.txt", "numlinksOutdegree.txt", "fdrTrimOutdegree.txt", "numlinksTrimOutdegree.txt"], "script": "countOutdegree.py", "parameters": [{"max": null, "min": null, "name": "resultMatrixFdr", "type": "inputFile", "defaultValue": "resultMatrixFdr.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "resultMatrixNumlinks", "type": "inputFile", "defaultValue": "resultMatrixNumlinks.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "fdrTrim", "type": "inputFile", "defaultValue": "fdrTrim.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "numlinksTrim", "type": "inputFile", "defaultValue": "numlinksTrim.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "fdrOutdegree", "type": "outputFile", "defaultValue": "fdrOutdegree.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "numlinksOutdegree", "type": "outputFile", "defaultValue": "numlinksOutdegree.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "fdrTrimOutdegree", "type": "outputFile", "defaultValue": "fdrTrimOutdegree.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "numlinksTrimOutdegree", "type": "outputFile", "defaultValue": "numlinksTrimOutdegree.txt", "fileExtension": ".txt"}], "isVisualization": false}, "5": {"name": "Barplot", "input": ["target.sif"], "nodeId": 20, "output": ["Barplot.json"], "script": "Barplot.py", "parameters": [{"max": null, "min": null, "name": "target", "type": "inputFile", "defaultValue": "target.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "output", "type": "outputFile", "defaultValue": "Barplot.json", "fileExtension": ".json"}, {"max": 100, "min": 5, "name": "Top Genes", "type": "int", "defaultValue": "10", "fileExtension": ""}], "isVisualization": true}, "6": {"name": "NeworkX", "input": ["target.sif"], "nodeId": 19, "output": ["NeworkX.json"], "script": "NetworkX.py", "parameters": [{"max": null, "min": null, "name": "target", "type": "inputFile", "defaultValue": "target.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "output", "type": "outputFile", "defaultValue": "NeworkX.json", "fileExtension": ".json"}, {"max": 200, "min": 10, "name": "Top Genes", "type": "int", "defaultValue": "50", "fileExtension": ""}], "isVisualization": true}, "7": {"name": "Heatmap", "input": ["expMatrix.csv", "pseudotime.csv", "target.sif"], "nodeId": 21, "output": ["Heatmap.json"], "script": "Pseudotime_heatmap.R", "parameters": [{"max": null, "min": null, "name": "expMatrix", "type": "inputFile", "defaultValue": "expMatrix.csv", "fileExtension": ".csv"}, {"max": null, "min": null, "name": "pseudotime", "type": "inputFile", "defaultValue": "pseudotime.csv", "fileExtension": ".csv"}, {"max": null, "min": null, "name": "target", "type": "inputFile", "defaultValue": "target.sif", "fileExtension": ".sif"}, {"max": null, "min": null, "name": "output", "type": "outputFile", "defaultValue": "Heatmap.json", "fileExtension": ".json"}, {"max": null, "min": 5, "name": "Top Genes", "type": "int", "defaultValue": "10", "fileExtension": ""}, {"max": 1000, "min": 50, "name": "Sample Size", "type": "int", "defaultValue": "100", "fileExtension": ""}], "isVisualization": true}, "8": {"name": "Run", "input": ["expMatrix.csv", "pseudotime.csv", "cellSelect.csv", "TF.txt"], "nodeId": 33, "output": ["resultMatrix.txt"], "script": "run_optional_tf.py", "parameters": [{"max": null, "min": null, "name": "expMatrix", "type": "inputFile", "defaultValue": "expMatrix.csv", "fileExtension": ".csv"}, {"max": null, "min": null, "name": "pseudotime", "type": "inputFile", "defaultValue": "pseudotime.csv", "fileExtension": ".csv"}, {"max": null, "min": null, "name": "cellSelect", "type": "inputFile", "defaultValue": "cellSelect.csv", "fileExtension": ".csv"}, {"max": null, "min": null, "name": "TF", "type": "optionalInputFile", "defaultValue": "TF.txt", "fileExtension": ".txt"}, {"max": null, "min": null, "name": "resultMatrix", "type": "outputFile", "defaultValue": "resultMatrix.txt", "fileExtension": ".txt"}, {"max": 32, "min": 1, "name": "number of devices", "type": "int", "defaultValue": "4", "fileExtension": ""}], "isVisualization": false}}');


--
-- Name: plugins_id_seq; Type: SEQUENCE SET; Schema: public; Owner: testdb
--

SELECT pg_catalog.setval('public.plugins_id_seq', 2, true);


--
-- Name: plugins plugins_pkey; Type: CONSTRAINT; Schema: public; Owner: testdb
--

ALTER TABLE ONLY public.plugins
    ADD CONSTRAINT plugins_pkey PRIMARY KEY (id);


--
-- Name: ix_plugins_id; Type: INDEX; Schema: public; Owner: testdb
--

CREATE INDEX ix_plugins_id ON public.plugins USING btree (id);


--
-- PostgreSQL database dump complete
--

